### PR TITLE
pkg5: wrap 'to modify' package list

### PR DESCRIPTION
**Pull request by github-kjsj**
##### SUMMARY
When executing against an OmniOS (r151030) node with only the Python 3.5.7 package installed, pkg5 module tasks fail due to differing behavior between Python 2.x and 3.x (if the Python 2 package is present, /usr/bin/python defaults to that, and everything works as expected). Wrapping filter() with a list() should cover both 2.x and 3.x.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
pkg5

##### ADDITIONAL INFORMATION
The trace returned when the module fails:
> Traceback (most recent call last):
>   File "<stdin>", line 113, in <module>
>   File "<stdin>", line 105, in _ansiballz_main
>   File "<stdin>", line 48, in invoke_module
>   File "/usr/lib/python3.5/imp.py", line 235, in load_module
>     return load_source(name, filename, file)
>   File "/usr/lib/python3.5/imp.py", line 170, in load_source
>     module = _exec(spec, sys.modules[name])
>   File "<frozen importlib._bootstrap>", line 626, in _exec
>   File "<frozen importlib._bootstrap_external>", line 697, in exec_module
>   File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
>   File "/tmp/ansible_pkg5_payload_lw9t_kg5/__main__.py", line 154, in <module>
>   File "/tmp/ansible_pkg5_payload_lw9t_kg5/__main__.py", line 88, in main
>   File "/tmp/ansible_pkg5_payload_lw9t_kg5/__main__.py", line 129, in ensure
> TypeError: can only concatenate list (not "filter") to list

### CODE REVIEW

**Comment by mavit** [comment]
LGTM

**Comment by github-kjsj** [comment]
ready_for_review

-- 
 Ken Smith
 kjsj@protogenoi.net

**Comment by github-kjsj** [comment]
bot_status

**Comment by github-kjsj** [comment]
ready_for_review

**Comment by pasztor** [comment]
> When executing against an OmniOS (r151030) node with only the Python 3.5.7 package installed, pkg5 module tasks fail due to differing behavior between Python 2.x and 3.x (if the Python 2 package is present, /usr/bin/python defaults to that, and everything works as expected). Wrapping filter() with a list() should cover both 2.x and 3.x.

Nice work! I've just spent hours to find out why does not pkg installation works on my system. Learned to debug ansible with pdb. Got to the same conclusion, and just wanted to send in the same fix. Before that, I checked if it's detected already.
Indeed it is.
Why is this trivial fix still not merged after almost half year?
I think, the next what I have to build is a pipeline to build an ansible from my own branch, and use that version :)

**Comment by github-kjsj** [comment]
@pasztor So I'm not the only one that ran into this! Unfortunately, I don't know why it hasn't been merged by a maintainer yet.

**Review by pyhalov — APPROVED** [review]

**Comment by pyhalov** [comment]
Stepped on the same issue on OpenIndiana

**Comment by mavit** [comment]
bot_status

**Comment by mavit** [comment]
>  Why is this trivial fix still not merged after almost half year?

It’s because there are far more open pull requests than the core Ansible team can cope with.

That said, `ansibot` should merge this for us if certain conditions are met.  As I understand it, we currently need:

1. The pull request to be [rebased](https://docs.ansible.com/ansible/devel/dev_guide/developing_rebasing.html) by the pull request submitter.
2. The pull request to be [reviewed](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md#when-will-your-pull-request-be-merged) by:
   - Other than me, one more packaging module maintainer or core team member.
   - Or, any three other people.

You can’t use GitHub’s review interface.  Instead you must issue the `shipit` command to the bot.

Hope this helps.

**Comment by mavit** [comment]
Oh, one more thing: there's been another repository reshuffle, so it turns out that this pull request now [needs to be resubmitted](https://github.com/ansible/ansibullbot/blob/master/docs/collection_migration.md#what-should-i-do-now) against https://github.com/ansible-collections/community.general/blob/main/plugins/modules/packaging/os/pkg5.py.  Sorry.

**Pull request closed without merging**
